### PR TITLE
Avoid re-render when clicking the bell in  "Display Farmer" and "Add farmer" components

### PR DESCRIPTION
**Pull request by daquinons**
# Description

When clicking on the notification bell a whole rerender occurs that makes the components reload the page. Improve the display to use passed props instead of requiring so many states inside of components.

## Checklist

Remove any items which are not applicable.

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by daquinons** [comment]
[Clicking the bell re-renders "Display Farmer" and "Add farmer" components](https://trello.com/c/LiudWHvF/163-clicking-the-bell-re-renders-display-farmer-and-add-farmer-components)

File: src/components/App/App.js [hunk 1]
```diff
@@ -180,14 +165,23 @@ function App() {
...
+              if (data.farmers) {
+                selectedFarmer = getFarmer(id);
+                if (!selectedFarmer) selectedFarmer = null;
+              }
+
+              return (
+                <DisplayFarmer
+                  {...props}
+                  farmer={selectedFarmer}
+                  getFarmer={getFarmer}
```
Comment (quansenB) [inline src/components/App/App.js]: As far as I can tell, this prop is not needed in DisplayFarmer

File: src/components/common/PageHeader/PageHeader.js [hunk 2]
```diff
@@ -16,14 +16,27 @@ const Span = styled.span`
...
+    if (visible) {
+      setVisible(!visible);
+    }
+  };
+
+  const toggleSideBar = () => {
+    setVisible(!visible);
+  };
+
+  React.useEffect(function setupListener() {
```
Comment (quansenB) [inline src/components/common/PageHeader/PageHeader.js]: image

Comment (quansenB) [reply src/components/common/PageHeader/PageHeader.js]: Didn't you forget the Dependencies Array?

File: src/components/App/App.js (hunk 1 above)
Comment (daquinons) [reply src/components/App/App.js]: Correct, I moved the function call into to the router and I forgot to remove the prop

File: src/components/common/PageHeader/PageHeader.js (hunk 2 above)
Comment (daquinons) [reply src/components/common/PageHeader/PageHeader.js]: Hahahah. Well is so sorcery that I learnt that its behaviour is different when there is no array than when the array is empty 🤯 If you set console logs just before the event is added and another one just before the event is removed you'll see it.

Comment (daquinons) [reply src/components/common/PageHeader/PageHeader.js]: To see how it plays out after the console logs you need to open the panel and click in anywhere to close it, that's when it's triggered.

**Pull request merged**